### ISSUE TITLE
privacy considerations for place_of_residence

### ISSUE THREAD

**Issue by jamwalla** [Recidiviz/recidiviz-data#769]
We need to be sure we don't store people's addresses, and generally try to prevent this field from identifying individuals. Below are some considerations:

Where can we scrub this information out?
 - in the scraper: the scraper writer has the most context on how addresses are formatted. But this puts a lot of trust in an individual for each scraper, and adds complexity for volunteers.
 - in the converter: we could use a parsing library and try to ensure that full addresses never enter our database in the first place.
- as a separate job on the database: we could use a maps api and have certainty that we are only storing county or zip, for example.

What should we store?
- city, state, zip: (requires parsing)
- zip only: this is pretty error-free, since we know what zip codes look like and could make the conversion at scrape time
- county only: (requires parsing)
- in/out of county (#415): the logic might be complicated, but this is the minimal amount of information necessary to make some analyses of geography

**Comment by jessex** [comment]
I personally think we should be storing facility information wherever it is attainable. At the very least, we should have it as an option on `sentence` to denote where the sentence is being served.

On the first larger comment, this is a tricky one. I'd argue for the location of the first option but the encapsulation of the third. To me, the appeal of the third option to me is that it would be a staging area where we could encapsulate all manner of anonymization/privacy-enhancing logic. It could scan certain fields for certain kinds of information we'd want to scrub, and run the appropriate scrubbing algorithm. But I think running this as a post-hoc job on the database is non-ideal since it opens up temporary attack vectors, requires additional compute and query time, and leads to lots of spurious querying and modification. It would be far more convenient to do this when you're performing ingest, and if you did build that sort of centralized privacy enhancer, you could just run that over each new `ingest_info` that is produced at scraper time.

**Comment by jamwalla** [comment]
#415

**Comment by jamwalla** [comment]
anecdotal data in conversation with @zdg2102:
the vast majority of people don't have place_of_residence.
of those that do, ~95% (800/15800) have 5 digits in them, so we could call those ZIP codes.

Some interesting other values:
```
HOMELESS <place>
AT LARGE <place>
HOMELESS AT LARGE
HENDRY CORRECTIONAL INSITUTIO IMMOKALEE FL
HENDRY CORRECTION INSTITUTION IMMOKALEE FL
SUWANEE CORRECTIONAL
216 SE CORRECTIONS WAY LAKE CITY FL 32025
945 N TEMPLE AVE/ BRADFORD CO JAIL APT C STARKE, FL 32091
FLORIDA STATE PRISON STARKS
UNKNOWN
RE ACT
```

@jessex suggests we could categorize some of this data as `PERMANENT`, `HOMELESS`, `TRANSIENT`, etc.

re: ZIP codes, are we ok with the confidence of "find 5 digits"? Is ZIP code sufficiently anonymous?

**Comment by jamwalla** [comment]
Josh found the [uszipcode](https://pypi.org/project/uszipcode/) package, and we talked with @ohinds. Here's the plan:
- Keep `place_of_residence` on IngestInfo
- Add a `resides_in_county` boolean to the schema, and use `uszipcode` to determine if the person lives in the county they are incarcerated in
- Add a `residency_status` enum to the schema, and search for strings to determine if the person's residency status is `PERMANENT`, `HOMELESS`/`HOUSELESS`, `IN CUSTODY` (the last if the place_of_residence is listed as another facility such as a jail or prison)

We'll double-check with @jkangbrown about all of this.

**Comment by jessex** [comment]
It's worth checking on how other organizations like the US Census may classify residency status. `transient` seems like something important to try to capture as distinct from `homeless`. Lots of communities are more transient, even in the US, and there are also refugees and migrants who you would likely want to give special consideration to.

**Comment by zdg2102** [comment]
Closed with #1231
